### DISCUSSION TITLE
Catch java.lang.System.exit() call

### DISCUSSION THREAD

**Discussion by rtgdk** [jpype-project/jpype#1428] [Announcements]
Is there a way to catch java.lang.System.exit() call so that the process does not stop abruptly ?
I am using JPype in a Django application and the connected java app calls System.exit in several methods causing the django server to stop.

**Comment by Thrameos** [comment]
I have not tested this myself but I would assume the correct solution is to use a JProxy to implement a SecurityManager and prevent the System.exit().

See https://stackoverflow.com/questions/5549720/how-to-prevent-calls-to-system-exit-from-terminating-the-jvm

**Comment by rtgdk** [comment]
JProxy implements interfaces but SecurityManager is a class. How to implement a class in JPype?

**Comment by Thrameos** [comment]
Hmm.  I will have to test that.  It should be possible to proxy a class.  One work around would be to just write a class that does the security exception and load it if the proxy does not work.

**Comment by Thrameos** [comment]
It is not possible to Proxy a class.  Currently the only way to handle this is to create a java class and load it to override the methods.  At some point in the future we could create byte code to extend a class from withing python but this it on a long time scale.

**Comment by Thrameos** [comment]
This one remains stuck on enhancement #420 which is a long term goal.

**Comment by Thrameos** [comment]
The extension api will complete this request.  Currently targeted for 1.7?
